processor/src/host/mod.rs

```diff
@@ -18,6 +21,17 @@ use handlers::DebugHandler;
 mod mast_forest_store;
 pub use mast_forest_store::{MastForestStore, MemMastForestStore};
 
+// ADVICE MAP MUTATIONS
+// ================================================================================================
+
+/// Any possible way an event can modify the advice map
+#[derive(Debug, PartialEq, Eq)]
+pub enum AdviceMutation {
+    ExtendStack { iter: Vec<Felt> },
+    ExtendMap { other: AdviceMap },
+    ExtendMerkleStore { iter: Vec<InnerNodeInfo> },
+}
```

Comment:
I would add doc comments to the variants to explain how they affect the advice provider.